### ISSUE TITLE
Enforce that ValueTuple types are structs

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#11689]
We seem to be gradually taking some direct or indirect dependencies on tuples being structs already - we allow nullable conversions, may consider ctor invocations equivalent to element-wise assignments into a local with default value, etc...

There is no point in having special support for a case where ValueTuple are classes and it is more robust to just require tuple underlying types to be structs.

**Comment by jcouv** [comment]
CC @AlekseyTs

**Comment by jcouv** [comment]
@VSadov Could you provide more examples where such enforcement would be useful?
Aleksey thought this could be more trouble than it's worth (because it is a costly check to do early).
Maybe we leave this as-is?

**Comment by AlekseyTs** [comment]
> Aleksey thought this could be more trouble than it's worth (because it is a costly check to do early).

Not only this is costly. It is also dangerous as it is likely to cause circularity during binding. I learned this the hard way while working on Non-nullable Reference Types. Checks whether a type is a value type should be postponed as much as possible.

**Comment by VSadov** [comment]
I think we should try.
We can always leave it as-is.

**Comment by VSadov** [comment]
It does not need to be a 100% comprehensive check.
The goal is to make it as obvious as possible that "reference ValueTuple" is not a supported scenario.

**Comment by AlekseyTs** [comment]
I think we should do the opposite, i.e. don't assume that it is a struct, or verify this assumption only where implementation makes such assumption (ideally there should be no places like that). So, before we start trying something, we should review places where the assumption is made and evaluate how important/need that assumption.
